stytra/experiments/camera_recording_experiment.py

```diff
@@ -1,61 +0,0 @@
-from multiprocessing import Event, Queue
-from stytra.collectors import FramerateQueueAccumulator
-
-from stytra.hardware.video.write import H5VideoWriter
-
-from stytra.experiments.tracking_experiments import CameraVisualExperiment
-from stytra.tracking.tracking_process import DispatchProcess
-
-
-class VideoRecordingExperiment(CameraVisualExperiment):
```

Comment:
I see the issue is that one could not make both a Tracking and Recording experiment. These functionalities should somehow be mix-ins that can get added to the default experiment class. I know it gets complicated with all the processes and connecting queues...
But for the recording case, I still think you could make an additional class rather than expanding `CameraVisualExperiment` to always have recording capabilities. It would have the camera queue, recording parameters, and timestamp information (for filenames) as inputs, a  `start` and `stop` interface,and a frame copy queue as output

You could have two versions of this class, one that is just a passthrough that just has the same queue reference as input and output.

This makes me think there has to be a more generic interface for frame-handling processes, that handle receiving paramters and passing data through (as data gets removed from multiprocessing queues when read)

Reply:
This was actually my suggestion (expanding `CameraVisualExperiment`), mainly b/c of the feeling those experiments classes are quite "monolitic" and not trivial to made in mix-in versions, so although I see that conceptually it could make sense to separate the two processes, for practical purposes I think that can be just methods of existing `CameraExperiment`. I would consider an additional class that always have recording capabilities, but it feels a bit of duplication of current  `CameraVisualExperiment` without recording. But no strong opinions there.

Your mention of generic interfaces for frame-handling processes is something that we discussed with @nvbln, I agree that having them would be nice and potentially simplify a bit the definition and mixability of those classes by cleaning up the hardwired queue wiring that we have at the moment - not sure it's something to be considered in the context of this PR

Reply:
I think having an additional class that is the same as `CameraVisualExperiment` but just always has recording capabilities would be a bit strange, but if this is the way to go forward I can implement it. For everything else I agree with what @vigji said.